localstack-core/localstack/utils/lambda_debug_mode/lambda_debug_mode_session.py

```diff
@@ -11,61 +14,135 @@
     load_lambda_debug_mode_config,
 )
 from localstack.utils.objects import singleton_factory
+from localstack.utils.threads import TMP_THREADS
 
 LOG = logging.getLogger(__name__)
 
 
 class LambdaDebugModeSession:
     _is_lambda_debug_mode: bool
+
+    _configuration_file_path: Optional[str]
+    _watch_thread: Optional[Thread]
+    _initialised_event: Optional[Event]
     _config: Optional[LambdaDebugModeConfig]
 
     def __init__(self):
         self._is_lambda_debug_mode = bool(LAMBDA_DEBUG_MODE)
-        self._configuration = self._load_lambda_debug_mode_config()
+
+        # Disabled Lambda Debug Mode state initialisation.
+        self._configuration_file_path = None
+        self._watch_thread = None
+        self._initialised_event = None
+        self._config = None
+
+        # Lambda Debug Mode is not enabled: leave as disabled state and return.
+        if not self._is_lambda_debug_mode:
+            return
+
+        # Lambda Debug Mode is enabled.
+        # Instantiate the configuration requirements if a configuration file is given.
+        self._configuration_file_path = LAMBDA_DEBUG_MODE_CONFIG_PATH
+        if not self._configuration_file_path:
+            return
+
+        # A configuration file path is given: initialised the resources to load and watch the file.
+
+        # Signal and block on first loading to ensure this is enforced from the very first
+        # invocation, as this module is not loaded at startup. The LambdaDebugModeConfigWatch
+        # thread will then take care of updating the configuration periodically and asynchronously.
+        # This may somewhat slow down the first upstream thread loading this module, but not
+        # future calls. On the other hand, avoiding this mechanism means that first Lambda calls
+        # occur with no Debug configuration.
+        self._initialised_event = Event()
+
+        self._watch_thread = Thread(
+            target=self._watch_logic, args=(), daemon=True, name="LambdaDebugModeConfigWatch"
+        )
+        TMP_THREADS.append(self._watch_thread)
```

Comment:
Appending it to `TMP_THREADS` accomplishes nothing here, as it is unstoppable.
Instead, as seen in the other comment, I would prefer shutting it down when the lambda service shuts down.

Reply:
I removed this